### PR TITLE
Ignore dead_code hints for weak-only null checks.

### PR DESCRIPTION
**Pull request by stereotype441**
## Description

When https://github.com/dart-lang/sdk/issues/41985 is fixed, the
analyzer will begin detecting dead code due to "unnecessary" null
checks.  Since we want to retain null checks in Flutter even when they
appear unnecessary (in order to preserve runtime behavior in weak
mode), we need to suppress these dead code hints.

## Related Issues

https://github.com/dart-lang/sdk/issues/41985 (PR must land before this issue can be fixed)

## Tests

N/A since this change does not affect behavior, it merely avoids dead code hints.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by stereotype441** [comment]
Side note to @a14n: the corresponding spec change (https://github.com/dart-lang/language/pull/1134) hasn't been reviewed yet.  To avoid unnecessary churn, I won't land this without approval on the spec change.

File: packages/flutter/lib/src/services/platform_channel.dart
Comment (Hixie) [inline packages/flutter/lib/src/services/platform_channel.dart:418]: nit: align the `:` with the `?` for clarity

**Comment by Hixie** [comment]
test exempt: temporary migration annotations

**Comment by Hixie** [comment]
Please do add some comments though that say what's going on.

**Comment by stereotype441** [comment]
> Please do add some comments though that say what's going on.

Done

**Review by a14n — COMMENTED** [review]
Reading the doc comment for `platform_channel.dart` `handler` type should be changed to be nullable. I will make a PR to change that.

**Comment by a14n** [comment]
#63057 has been merged.

File: packages/flutter/lib/src/services/platform_channel.dart
Comment (stereotype441) [reply packages/flutter/lib/src/services/platform_channel.dart]: Good point.  Looks like this is moot now, since @a14n made the type of `handler` nullable.  I've reverted this part of the PR.

**Comment by stereotype441** [comment]
> Reading the doc comment for `platform_channel.dart` `handler` type should be changed to be nullable. I will make a PR to change that.

@a14n thanks for noticing this.  It looks like the type of `handler` in `setMockMessageHandler` should be made nullable too.  I went ahead and did that in this PR; if you'd rather put that change in a separate PR let me know and I can rebase this one on top of yours.

**Review by a14n — APPROVED** [review]
LGTM

**Pull request merged**
